src/praisonai-agents/praisonaiagents/memory/core.py

```
@@ -62,16 +62,14 @@ def store_short_term(self, content: str, metadata: Optional[Dict] = None, qualit
         
-        # Backward compatibility: Also store in SQLite if not using SQLite adapter
-        if hasattr(self, '_sqlite_adapter') and self._sqlite_adapter != getattr(self, 'memory_adapter', None):
+        # Only use SQLite fallback if primary storage failed completely
+        if not memory_id and hasattr(self, '_sqlite_adapter') and self._sqlite_adapter != getattr(self, 'memory_adapter', None):
             try:
-                fallback_id = self._sqlite_adapter.store_short_term(content, metadata=clean_metadata, **kwargs)
-                if not memory_id:
-                    memory_id = fallback_id
+                memory_id = self._sqlite_adapter.store_short_term(content, metadata=clean_metadata, **kwargs)
+                self._log_verbose(f"Stored in SQLite STM as fallback: {content[:100]}...")
             except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)